src/emit/soroban/mod.rs

```diff
@@ -294,8 +296,20 @@ impl SorobanTarget {
             .build_call(storage_initializer, &[], "storage_initializer")
             .unwrap();
 
+        // call the user defined constructor (if any)
+        if let Some(cfg_no) = constructor_cfg_no {
+            let constructor = binary.functions[cfg_no];
+            let constructor_name = constructor.get_name().to_str().unwrap();
+            binary
+                .builder
+                .build_call(constructor, &[], constructor_name)
+                .unwrap();
+        }
```

Comment:
We can set `cfg.params` to the parameters of the constructor `FunctionValue`'s params. This will likely include a mapping that looks something like:
```rust
let mut params = vec![];
for param in constructor.get_params() {
	let param = match param.get_type() {
		BasicTypeEnum::IntType(int_type) => {
			let ty = match int_type.get_bit_width() {
				1 => ast::Type::Bool,
				32 => ast::Type::Uint(32),
				64 => ast::Type::Uint(64),
				_ => unreachable!(),
			};
			ast::Parameter::new_default(ty)
		}
		_ => unreachable!(),
	};
	params.push(param);
}
cfg.params = sync::Arc::new(params);
```
This won’t work because we still need to pass the arguments to `build_call()` (on line 296), which, of course, we don’t know.